### PR TITLE
send comments

### PR DESCRIPTION
**Pull request by joelhooks**

### CODE REVIEW

File: src/components/pages/lessons/comments/comment-field/index.tsx [hunk 1]
```diff
@@ -8,9 +8,13 @@ import {Formik} from 'formik'
 
 type CommentFieldProps = {
   url?: string
+  onSubmit?: any
 }
 
-const CommentField: FunctionComponent<CommentFieldProps> = ({url}) => {
+const CommentField: FunctionComponent<CommentFieldProps> = ({
+  url,
+  onSubmit = () => {},
```
Comment (joelhooks) [inline src/components/pages/lessons/comments/comment-field/index.tsx:16]: we want to handle the event above

File: src/components/pages/lessons/comments/comments/index.tsx [hunk 2]
```diff
@@ -14,31 +16,43 @@ const Comments: React.FunctionComponent<CommentsProps> = ({
...
+  const [comments, setComments] = React.useState(lesson.comments)
+  const {slug} = lesson
   const commentsAvailable =
     comments?.some((comment: any) => comment.state === 'published') ?? false
+
+  const handleCommentSubmission = async (comment: string) => {
+    const newComment = await saveCommentForLesson(slug, {
+      comment,
+    })
+    setComments([...comments, newComment])
```
Comment (joelhooks) [inline src/components/pages/lessons/comments/comments/index.tsx:29]: add the comment after it saves

File: src/lib/lesson-comments.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import axios from 'utils/configured-axios'
+
+export type Comment = {
+  title?: string
+  comment: string
+  role?: string
+}
+
+export async function saveCommentForLesson(slug: string, comment: Comment) {
+  return await axios
```
Comment (joelhooks) [inline src/lib/lesson-comments.ts:10]: persist the new comment

**Review by joelhooks — COMMENTED** [review]
this should be ready to go!

**Comment by joelhooks** [comment]
won't work until this is merged https://github.com/eggheadio/egghead-rails/commit/8442c03d21f8af2c20c11ab27088ec50a14e0fb5

**Review by DrShpongle — APPROVED** [review]

**Comment by DrShpongle** [comment]
Cool, thanks for figuring it out 🙌

**Pull request merged**